### ISSUE TITLE
Wave 2.4.0 crashes after GoodnessOfFit Analysis

### ISSUE THREAD

**Issue by TobiasRosskopf** [bluemodel/BlueM.Wave#105]
Since BlueM.Wave 2.4.0 the **GoodnessOfFit** Analysis does not work anymore (in 2.3.1 it's fine). Maybe problems with the new result table. The other new revised analysis functions (AnnualStatistics, Histogram, MonthlyStatistics) are runing as expected. System is Windows 10 Pro (22H2).

![Wave_Bug_01](https://github.com/bluemodel/BlueM.Wave/assets/42942019/24fe93bf-13e3-4a7e-997b-199363de6055)
![Wave_Bug_02](https://github.com/bluemodel/BlueM.Wave/assets/42942019/10094da0-972c-4d9e-9a2c-b9598b753f4f)
![Wave_Bug_04](https://github.com/bluemodel/BlueM.Wave/assets/42942019/3d7aa373-4ad6-413d-b773-537b86e89ae5)
![Wave_Bug_03](https://github.com/bluemodel/BlueM.Wave/assets/42942019/5da83c9d-4ab0-43dd-aa67-b57406a4957f)

**Complete stack trace:**
```
Informationen über das Aufrufen von JIT-Debuggen
anstelle dieses Dialogfelds finden Sie am Ende dieser Meldung.

************** Ausnahmetext **************
System.OverflowException: Überlauffehler.
   bei System.Drawing.Graphics.CheckErrorStatus(Int32 status)
   bei System.Drawing.Graphics.DrawLine(Pen pen, Int32 x1, Int32 y1, Int32 x2, Int32 y2)
   bei Steema.TeeChart.Drawing.Graphics3DGdiPlus.LineTo(Int32 x, Int32 y)
   bei Steema.TeeChart.Drawing.Graphics3D.LineTo(Int32 x, Int32 y, Int32 z)
   bei Steema.TeeChart.Styles.CustomPolar.InternalDrawValue(Int32 valueIndex, Int32 X, Int32 Y)
   bei Steema.TeeChart.Styles.CustomPolar.nlo(Int32 eqj)
   bei Steema.TeeChart.Styles.Series.DoDrawIndex(Int32 Index, Boolean CanID)
   bei Steema.TeeChart.Styles.Series.Draw()
   bei Steema.TeeChart.Styles.CustomPolar.Draw()
   bei Steema.TeeChart.Styles.Series.DrawSeries()
   bei Steema.TeeChart.Chart.aup(Graphics3D eqj, Int32 eqk, Int32 eql, Int32 eqm)
   bei Steema.TeeChart.Chart.aun(Graphics3D eqj)
   bei Steema.TeeChart.Chart.auq(Graphics eqj, Boolean eqk)
   bei Steema.TeeChart.TChart.Draw(Graphics g)
   bei Steema.TeeChart.TChart.OnPaint(PaintEventArgs pe)
   bei System.Windows.Forms.Control.PaintWithErrorHandling(PaintEventArgs e, Int16 layer)
   bei System.Windows.Forms.Control.WmPaint(Message& m)
   bei System.Windows.Forms.Control.WndProc(Message& m)
   bei System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Geladene Assemblys **************
mscorlib
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4614.0 built by: NET48REL1LAST_B.
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll.
----------------------------------------
Wave
    Assembly-Version: 2.4.0.21001.
    Win32-Version: 2.4.0.21001.
    CodeBase: file:///p:/Prgm/UTILS/BlueM/BlueM.Wave_v2.4.0/Wave.exe.
----------------------------------------
System.Windows.Forms
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4550.0 built by: NET48REL1LAST_C.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll.
----------------------------------------
System
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4605.0 built by: NET48REL1LAST_C.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll.
----------------------------------------
System.Drawing
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4390.0 built by: NET48REL1LAST_C.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll.
----------------------------------------
System.Core
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4605.0 built by: NET48REL1LAST_C.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll.
----------------------------------------
TeeChart
    Assembly-Version: 4.2022.8.23.
    Win32-Version: 4.2022.8.23.
    CodeBase: file:///p:/Prgm/UTILS/BlueM/BlueM.Wave_v2.4.0/TeeChart.DLL.
----------------------------------------
Microsoft.VisualBasic
    Assembly-Version: 10.0.0.0.
    Win32-Version: 14.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.VisualBasic/v4.0_10.0.0.0__b03f5f7f11d50a3a/Microsoft.VisualBasic.dll.
----------------------------------------
System.Configuration
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4190.0 built by: NET48REL1LAST_B.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll.
----------------------------------------
System.Xml
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll.
----------------------------------------
?
    Assembly-Version: 0.0.0.0.
    Win32-Version: 4.2022.8.23.
    CodeBase: file:///p:/Prgm/UTILS/BlueM/BlueM.Wave_v2.4.0/TeeChart.dll.
----------------------------------------
System.Windows.Forms.resources
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms.resources/v4.0_4.0.0.0_de_b77a5c561934e089/System.Windows.Forms.resources.dll.
----------------------------------------
System.Data
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4584.0 built by: NET48REL1LAST_B.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_64/System.Data/v4.0_4.0.0.0__b77a5c561934e089/System.Data.dll.
----------------------------------------
System.Net.Http
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Net.Http/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Net.Http.dll.
----------------------------------------
Accessibility
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Accessibility/v4.0_4.0.0.0__b03f5f7f11d50a3a/Accessibility.dll.
----------------------------------------
PydioHandler
    Assembly-Version: 1.3.1.20209.
    Win32-Version: 1.3.1.20209.
    CodeBase: file:///C:/Program%20Files%20(x86)/PydioSync/shell/dll/PydioHandler.DLL.
----------------------------------------
SharpShell
    Assembly-Version: 2.2.0.0.
    Win32-Version: 2.2.0.0.
    CodeBase: file:///C:/Program%20Files%20(x86)/PydioSync/shell/dll/SharpShell.DLL.
----------------------------------------
System.ComponentModel.Composition
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.ComponentModel.Composition/v4.0_4.0.0.0__b77a5c561934e089/System.ComponentModel.Composition.dll.
----------------------------------------
Microsoft.CSharp
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/Microsoft.CSharp/v4.0_4.0.0.0__b03f5f7f11d50a3a/Microsoft.CSharp.dll.
----------------------------------------
System.Web.Extensions
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4494.0.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Web.Extensions/v4.0_4.0.0.0__31bf3856ad364e35/System.Web.Extensions.dll.
----------------------------------------
System.Numerics
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Numerics/v4.0_4.0.0.0__b77a5c561934e089/System.Numerics.dll.
----------------------------------------
mscorlib.resources
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/mscorlib.resources/v4.0_4.0.0.0_de_b77a5c561934e089/mscorlib.resources.dll.
----------------------------------------
System.Web
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4494.0 built by: NET48REL1LAST_B.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_64/System.Web/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Web.dll.
----------------------------------------
System.Design
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Design/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Design.dll.
----------------------------------------
System.Drawing.resources
    Assembly-Version: 4.0.0.0.
    Win32-Version: 4.8.4084.0 built by: NET48REL1.
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing.resources/v4.0_4.0.0.0_de_b03f5f7f11d50a3a/System.Drawing.resources.dll.
----------------------------------------

************** JIT-Debuggen **************
Um das JIT-Debuggen (Just-In-Time) zu aktivieren, muss in der
Konfigurationsdatei der Anwendung oder des Computers
(machine.config) der jitDebugging-Wert im Abschnitt system.windows.forms festgelegt werden.
Die Anwendung muss mit aktiviertem Debuggen kompiliert werden.

Zum Beispiel:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

Wenn das JIT-Debuggen aktiviert ist, werden alle nicht behandelten
Ausnahmen an den JIT-Debugger gesendet, der auf dem
Computer registriert ist, und nicht in diesem Dialogfeld behandelt.
```

**Error message in `Main.vb` when running in Visual Studio 2022:**

![Wave_Bug_05](https://github.com/bluemodel/BlueM.Wave/assets/42942019/401254b0-d336-457b-a364-70454768836a)

**Comment by jamaa** [comment]
thanks for the detailed bug report, I'll look into it.

**Comment by jamaa** [comment]
I think your exception originates from the result chart, and not the result table. However, I can't reproduce it using my test data. Is it possible to share the two time series that you used or save the chart as a TEN file and attach it here?

**Comment by jamaa** [comment]
I was able to reproduce the error using two series with the unit "mm", which causes the volume error to be calculated as NaN, which crashed the chart. This has been fixed in 1e19d2543736784c6f4d3e074d557778c684b41b.

However, in your image, the volume error is not NaN. But maybe one of the other result values was NaN? Can you test whether the issue is resolved for you using the latest master branch?

**Comment by jamaa** [comment]
The fix will be shipped with v2.4.2
